### PR TITLE
dev-libs/xmlsec: add upstream libressl patch

### PR DESCRIPTION
**Pull request by orbea**
These patches add compatiblity for LibreSSL.

Bug: https://bugs.gentoo.org/903001
Upstream-PR: https://github.com/lsh123/xmlsec/pull/456
Upstream-Commit: https://github.com/lsh123/xmlsec/commit/c5469cfc8443c57a25a8783f0bd669f71e29bb04
Upstream-PR: https://github.com/lsh123/xmlsec/pull/654
Upstream-Commit: https://github.com/lsh123/xmlsec/commit/dfdf981f3522e4059170b504fb6fd40b37c9d70f
Upstream-Issue: https://github.com/lsh123/xmlsec/issues/665
Upstream-PR: https://github.com/lsh123/xmlsec/pull/666
Upstream-Commit: https://github.com/lsh123/xmlsec/commit/1ee1754c5ab8f0071adbde92d3a007729df7c5a7
Upstream-PR: https://github.com/lsh123/xmlsec/pull/667
Upstream-Commit: https://github.com/lsh123/xmlsec/commit/c9b0dcd01af1ecaed828269b734861cb93edeae3

### CODE REVIEW

**Comment by a17r** [comment]
> The older xmlsec version is required for stable LibreOffice

7.5.3.2 stabilisation is already pending so this is irrelevant.

**Comment by orbea** [comment]
Without trying to be rude what you seem to be clear about is that any excuse to add obstacles for libressl will be taken.

> As far as xmlsec is concerned, it seems this fix will be in soon to be released 1.3.1 anyway?

>= 1.3.0 has additional issues which I am still working on as documented in the upstream PR.

**Comment by orbea** [comment]
Added patches for 1.3.0 too courtesy of upstream.

File: dev-libs/xmlsec/xmlsec-1.2.37.ebuild [hunk 1]
```diff
@@ -36,6 +36,10 @@ BDEPEND="virtual/pkgconfig
 		)
 	)"
 
+PATCHES=(
```
Comment (thesamesam) [inline dev-libs/xmlsec/xmlsec-1.2.37.ebuild:39]: You're touching the build system in this patch => need to inherit autotools & inherit eautoreconf.

Comment (orbea) [reply dev-libs/xmlsec/xmlsec-1.2.37.ebuild:39]: Wrong line? The patch for 1.2.37 only touches `src/openssl/openssl_compat.h` and 1.3.0-r1 already inherits autotools and uses `eautoreconf`.

https://github.com/gentoo/gentoo/blob/b0eb525e75e849d0b9dc9bfe58d66838089dfa3e/dev-libs/xmlsec/xmlsec-1.3.0-r1.ebuild#L55

Comment (thesamesam) [reply dev-libs/xmlsec/xmlsec-1.2.37.ebuild:39]: yes, sorry!

**Comment by thesamesam** [comment]
I've dropped the 1.3.0* changes as 1.3.1 is out now and in tree, which should fix that.

I've also moved the 1.2.37 stuff into a ~arch revbump as it should get some testing as there's <... deps in tree.

Thank you!

**Pull request closed without merging**